src/Analytics/Adapter.php

```diff
@@ -25,6 +25,15 @@ abstract class Adapter
      */
     abstract public function getName(): string;
 
+    /**
+     * Adapter constructor.
+     * 
+     * @param string $configuration
+     * 
+     * @return Adapter
+     */
+    abstract public function __construct(string $configuration);
```

Comment:
I don't really like this approach:
- Too easy to make mistakes
- Very hard to understand the format
- A lot of extra code to handle the options

It would be better to try and have a proper signature with all must have options we need to initializing the instance. If we require a lot of extra options we can add `setX` methods for less important options.